File: DuoShuo/import.php
<?php

// 读取 json 文件并转换成 php 数组
$json = file_get_contents("./export.json");
$data = json_decode($json, true);

// 文章数据
$threads = $data['threads'];	
// 评论数据
$posts = $data['posts'];	

// 多说的文章ID(thread_id)与 typecho 的文章ID(cid) 对应关系
$threadIdRelationCid = [];
foreach ($threads as $item) {
	$threadIdRelationCid[$item['thread_id']] = $item['thread_key'];
}

// 假设评论表 coid 小于10001，此处从1001开始自增，请根据实际最大值修改
$coid = 10001;
// 多说的评论ID(post_id) 与 typecho 的评论ID(coid) 对应关系
$postIdRelationCoid = [];
foreach ($posts as $item) {
	$postIdRelationCoid[$item['post_id']] = $coid++;
}

// 拼成多条 insert sql语句
$sql = '';
foreach ($posts as $item) {
	$coid = $postIdRelationCoid[$item['post_id']];
	$cid = $threadIdRelationCid[$item['thread_id']];
	$created = strtotime($item['created_at']);
	$author = $item['author_name'] ?: '';
	$mail = $item['author_email'] ?: '';
	$url = $item['author_url'] ?: '';
	$ip = $item['ip'];
	$text = $item['message'];
	$parent = 0;
	if (is_array($item['parents'])) {
		$parent = $postIdRelationCoid[$item['parents'][0]];
	}

	$sql .= "INSERT INTO `typecho_comments` 
(`coid`, `cid`, `created`, `author`, `authorId`, `ownerId`, `mail`, `url`, `ip`, `agent`, `text`, `type`, `status`, `parent`) VALUES
({$coid}, {$cid}, {$created}, '{$author}', 0, 1, '{$mail}', '{$url}', '{$ip}', NULL, '{$text}', 'comment', 'approved', $parent);\n";
}

// 将 sql 写入文件中
file_put_contents("./insert.sql", $sql);

echo "end \n";

/*
INSERT INTO `typecho_comments` 
(`coid`, `cid`, `created`, `author`, `authorId`, `ownerId`, `mail`, `url`, `ip`, `agent`, `text`, `type`, `status`, `parent`) VALUES
(8, 402, 1493430000, 'sean', 0, 1, 'json_vip@163.com', NULL, '::1', 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36', '赞的回复', 'comment', 'approved', 6);
*/

File: DuoShuo/import.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"os"
	"time"
)

// 多说文章数据struct
type Threads struct {
	// 和json字段格式不一样 就要指定
	ThreadId  int    `json:"thread_id"`  // 多说文章ID
	ThreadKey string `json:"thread_key"` // typecho文章ID
}

// 多说评论数据struct
type Posts struct {
	PostId      int    `json:"post_id"`    // 多说评论ID
	ThreadId    int    `json:"thread_id"`  // 多说文章ID
	CreatedAt   string `json:"created_at"` // 创建日期
	AuthorName  string `json:"author_name"`
	AuthorEmail string `json:"author_email"`
	AuthorUrl   string `json:"author_url"`
	Ip          string
	Message     string
	Parents     []int
}

// 多说最终返回数据struct
type Response struct {
	Generator string    //`json:"generator"`
	Version   string    //`json:"version"`
	Threads   []Threads //`json:"threads"`
	Posts     []Posts
}

func main() {

	resp, err := readFile("export.json")
	if err != nil {
		fmt.Println("readFile:", err.Error())
		return
	}

	//var threadIdRelationCid map[int]string
	threadIdRelationCid := make(map[int]string)
	for _, item := range resp.Threads {
		threadIdRelationCid[item.ThreadId] = item.ThreadKey
	}

	coid := 1001
	postIdRelationCoid := make(map[int]int)
	for _, item := range resp.Posts {
		postIdRelationCoid[item.PostId] = coid
		coid += 1
	}

	sql := ""
	for _, item := range resp.Posts {
		tcoid := postIdRelationCoid[item.PostId]
		cid := threadIdRelationCid[item.ThreadId]
		t, _ := time.Parse("2006-01-02T15:04:05Z07:00", item.CreatedAt)
		created := t.Unix()
		author := item.AuthorName
		mail := item.AuthorEmail
		url := item.AuthorUrl
		ip := item.Ip
		text := item.Message
		parent := 0
		if len(item.Parents) > 0 {
			parent = postIdRelationCoid[item.Parents[0]]
		}

		sql += "INSERT INTO `typecho_comments` (`coid`, `cid`, `created`, `author`, `authorId`, `ownerId`, `mail`, `url`, `ip`, `agent`, `text`, `type`, `status`, `parent`) VALUES"
		sql += fmt.Sprintf("(%d, %s, %d, '%s', 0, 1, '%s', '%s', '%s', '', '%s', 'comment', 'approved', %d);\n", tcoid, cid, created, author, mail, url, ip, text, parent)
	}

	//fmt.Println(sql)
	writeFile("insert.sql", sql)
	fmt.Println("end")
}

// 读取文件的内容并返回
func readFile(filename string) (resp Response, err error) {

	bytes, err := ioutil.ReadFile(filename)
	if err != nil {
		return
	}

	if err := json.Unmarshal(bytes, &resp); err != nil {
		fmt.Println(err)
	}

	return resp, err
}

// 向文件写入内容
func writeFile(filename string, text string) {

	fout, err := os.Create(filename)
	if err != nil {
		fmt.Println(filename, err)
		return
	}
	defer fout.Close()

	fout.WriteString(text)
}

// func read() {
// 	filename := "export.json"
// 	fout, err := os.Open(filename)
// 	if err != nil {
// 		fmt.Println(filename, err)
// 		return
// 	}
// 	defer fout.Close()

// 	var jsonStr string
// 	n, _ := fout.Read(jsonStr)

// 	fmt.Println(n, jsonStr)
// }
